File: 이훈일/Programmers/Level2/모음사전.java
class Solution {
    String[] words = new String[]{"A", "E", "I", "O", "U"};
    StringBuilder sb = new StringBuilder();
    String nWord;
    int count, answer;
    
    public int solution(String word) {
        answer = 0;
        count = 0;
        nWord = word;
        dfs(0);
        return answer;
    }
    
    public void dfs(int depth) {
        if(depth == 5) {
            return;
        }
        
        for(int i = 0; i < 5; i++) {
            sb.append(words[i]);
            count++;
            if(sb.toString().equals(nWord)) {
                answer = count;
                return;
            }
            dfs(depth+1);
            sb.deleteCharAt(depth);
        }
    }
}

File: 조준장/level2/모음사전.kt
class Solution {
    val alphabet = arrayListOf("A", "E", "I", "O", "U")
    val words = ArrayList<String>()
    
    fun solution(word: String): Int {
        var answer = 0
        dfs("")
        words.indices.forEach{
            if (words[it] == word){
                answer = it
                return@forEach
                
            } 
        }
        return answer
    }
    
    fun dfs(word : String){
        if (word.length > 5) return
        words.add(word)
        
        alphabet.forEach{
            dfs(word+it)
        }
    }
    
    
}
